### PR TITLE
Svc file split

### PR DESCRIPTION
**Pull request by pkalever**
### What does this PR achieve? Why do we need it?

Split the code from

rpc/
├── block_svc_routines.c

to

rpc/
├── block_common.h
├── block_create.c
├── block_delete.c
├── block_genconfig.c
├── block_info.c
├── block_list.c
├── block_modify.c
├── block_replace.c
├── block_svc_routines.c
├── block_version.c

### Notes for the reviewer

This will help us further organize the code better.

This PR on top of on https://github.com/gluster/gluster-block/pull/248

### CODE REVIEW

File: rpc/block_common.h [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  Copyright (c) 2016 Red Hat, Inc. <http://www.redhat.com>
+  This file is part of gluster-block.
+
+  This file is licensed to you under your choice of the GNU Lesser
+  General Public License, version 3 or any later version (LGPLv3 or
+  later), or the GNU General Public License, version 2 (GPLv2), in all
+  cases as published by the Free Software Foundation.
+*/
+
+
```
Comment (lxbsz) [inline rpc/block_common.h]: There should have the header file's multiple contained check:
```
# ifndef __BLOCK_COMMON_H
# define __BLOCK_COMMON_H

# endif
```

File: rpc/block_create.c [hunk 2]
```diff
@@ -0,0 +1,1151 @@
...
+  if (sptr && *(sptr + strlen(sptr) - 1) == '.') {
+    *(sptr + strlen(sptr) - 1) = '\0';
+  }
+
+  return sptr;
+}
+
+
+void
+removeDuplicateSubstr(char **line)
```
Comment (lxbsz) [inline rpc/block_create.c]: Should this be static ?

File: rpc/block_create.c [hunk 3]
```diff
@@ -0,0 +1,1151 @@
...
+  }
+  GB_FREE(args);
+  GB_FREE(tid);
+
+  return ret;
+}
+
+
+blockResponse *
+block_create_common(blockCreate *blk, char *control, char *volServer, char *prio_path)
```
Comment (lxbsz) [inline rpc/block_create.c]: Static ?

File: rpc/block_create.c [hunk 4]
```diff
@@ -0,0 +1,1151 @@
...
+    }
+  }
+
+  convertTypeCreate2ToCreate(blk, &blk_v1);
+
+  return block_create_common(&blk_v1, control, volServer, blk->prio_path);
+}
+
+void
+blockCreateCliFormatResponse(struct glfs *glfs, blockCreateCli *blk,
```
Comment (lxbsz) [inline rpc/block_create.c]: static ?

File: rpc/block_delete.c [hunk 5]
```diff
@@ -0,0 +1,521 @@
...
+  GB_FREE(args);
+  GB_FREE(tid);
+  GB_FREE(info_new);
+
+  return ret;
+}
+
+
+void
+blockDeleteCliFormatResponse(blockDeleteCli *blk, int errCode, char *errMsg,
```
Comment (lxbsz) [inline rpc/block_delete.c]: static ?

File: rpc/block_info.c [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+  later), or the GNU General Public License, version 2 (GPLv2), in all
+  cases as published by the Free Software Foundation.
+*/
+
+
+# include  "block_common.h"
+
+
+void
+blockInfoCliFormatResponse(blockInfoCli *blk, int errCode,
```
Comment (lxbsz) [inline rpc/block_info.c]: Static ?

File: rpc/block_replace.c [hunk 7]
```diff
@@ -0,0 +1,854 @@
...
+# include  "block_common.h"
+
+# define   GB_GET_PORTAL_TPG    "targetcli /iscsi/" GB_TGCLI_IQN_PREFIX \
+                                "'%s' ls | grep -e tpg -e '%s' | grep -B1 '%s' | grep -o 'tpg\\w'"
+# define   GB_CHECK_PORTAL      "targetcli /iscsi/" GB_TGCLI_IQN_PREFIX \
+                                "'%s' ls | grep '%s' > " DEVNULLPATH
+
+
+void
+blockRemoteReplaceRespFree(blockRemoteReplaceResp *resp)
```
Comment (lxbsz) [inline rpc/block_replace.c]: static ?

File: rpc/block_replace.c [hunk 8]
```diff
@@ -0,0 +1,854 @@
...
+
+  blockRemoteRespFree(resp->cop);
+  blockRemoteRespFree(resp->dop);
+  blockRemoteRespFree(resp->rop);
+  GB_FREE(resp);
+}
+
+
+void *
+glusterBlockReplacePortalRemote(void *data)
```
Comment (lxbsz) [inline rpc/block_replace.c]: static ?

File: rpc/block_replace.c [hunk 9]
```diff
@@ -0,0 +1,854 @@
...
+  }
+  args->exit = ret;
+
+  GB_FREE (errMsg);
+  return NULL;
+}
+
+
+int
+glusterBlockReplaceNodeRemoteAsync(struct glfs *glfs, blockReplaceCli *blk,
```
Comment (lxbsz) [inline rpc/block_replace.c]: static ?

File: rpc/block_replace.c [hunk 10]
```diff
@@ -0,0 +1,854 @@
...
+  GB_FREE(robj);
+  GB_FREE(args);
+  blockRemoteReplaceRespFree(reply);
+
+  return ret;
+}
+
+
+void
+blockReplaceNodeCliFormatResponse(blockReplaceCli *blk, int errCode, char *errMsg,
```
Comment (lxbsz) [inline rpc/block_replace.c]: static ?

File: rpc/block_version.c [hunk 11]
```diff
@@ -0,0 +1,379 @@
...
+  later), or the GNU General Public License, version 2 (GPLv2), in all
+  cases as published by the Free Software Foundation.
+*/
+
+
+# include  "block_common.h"
+
+
+bool *
+glusterBlockBuildMinCaps(void *data, operations opt)
```
Comment (lxbsz) [inline rpc/block_version.c]: static ?

File: rpc/block_version.c [hunk 12]
```diff
@@ -0,0 +1,379 @@
...
+  case VERSION_SRV:
+  case GENCONFIG_SRV:
+    break;
+  }
+
+  return minCaps;
+}
+
+void *
+glusterBlockCapabilitiesRemote(void *data)
```
Comment (lxbsz) [inline rpc/block_version.c]: static ?

File: rpc/block_version.c [hunk 13]
```diff
@@ -0,0 +1,379 @@
...
+
+  args->exit = ret;
+
+  return NULL;
+}
+
+
+/* function to imitate caps of older gluster-block versions */
+gbCapResp *
+glusterBlockMimicOldCaps(void)
```
Comment (lxbsz) [inline rpc/block_version.c]: static ?

File: rpc/block_modify.c [hunk 14]
```diff
@@ -0,0 +1,1143 @@
...
+  later), or the GNU General Public License, version 2 (GPLv2), in all
+  cases as published by the Free Software Foundation.
+*/
+
+
+# include  "block_common.h"
+
+
+void *
+glusterBlockModifyRemote(void *data)
```
Comment (lxbsz) [inline rpc/block_modify.c]: static ?

File: rpc/block_modify.c [hunk 15]
```diff
@@ -0,0 +1,1143 @@
...
+  }
+  GB_FREE(errMsg);
+  args->exit = ret;
+
+  return NULL;
+}
+
+
+void *
+glusterBlockModifySizeRemote(void *data)
```
Comment (lxbsz) [inline rpc/block_modify.c]: static ?

File: rpc/block_genconfig.c [hunk 16]
```diff
@@ -0,0 +1,432 @@
...
+
+  snprintf(iqn, 128, "%s%s", GB_TGCLI_IQN_PREFIX, info->gbid);
+  json_object_object_add(tg_obj, "wwn", GB_JSON_OBJ_TO_STR(iqn[0]?iqn:NULL));
+
+  return tg_obj;
+}
+
+
+struct json_object *
+getSoObj(char *block, MetaInfo *info, blockGenConfigCli *blk)
```
Comment (lxbsz) [inline rpc/block_genconfig.c]: static?

File: rpc/block_genconfig.c [hunk 17]
```diff
@@ -0,0 +1,432 @@
...
+
+  json_object_object_add(tpg_obj, "tag", json_object_new_int(tag));
+  // }  Tpg Object close
+
+  return tpg_obj;
+}
+
+
+struct json_object *
+getTgObj(char *block, MetaInfo *info, blockGenConfigCli *blk)
```
Comment (lxbsz) [inline rpc/block_genconfig.c]: static?

File: rpc/block_genconfig.c [hunk 18]
```diff
@@ -0,0 +1,432 @@
...
+
+
+# include  "block_common.h"
+
+# define   GB_ALUA_AO_TPG_NAME          "glfs_tg_pt_gp_ao"
+# define   GB_ALUA_ANO_TPG_NAME         "glfs_tg_pt_gp_ano"
+
+
+struct json_object *
+getTpgObj(char *block, MetaInfo *info, blockGenConfigCli *blk, char *portal, int tag)
```
Comment (lxbsz) [inline rpc/block_genconfig.c]: static ?

**Comment by pkalever** [comment]
@lxbsz please take a look.

**Comment by pkalever** [comment]
Merged now. Thanks!

**Pull request merged**